### PR TITLE
Scraper Procedure

### PR DESCRIPTION
**Pull request by Meggie-H**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,2 +1,91 @@
-console.log("Hello World");
-console.log(process.env.ELASTIC_CONNECTION_STRING);
+import { chromium } from "playwright";
+import ProductInterface from "./models/models";
+
+(async () => {
+	console.log("\nSetting up web scraper\n");
```
Comment (yanevdw) [inline src/index.ts:5]: Please remove.

File: src/index.ts [hunk 2]
```diff
@@ -1,2 +1,91 @@
...
+	const scrapedProducts: ProductInterface[] = [];
+	const browser = await chromium.launch();
+	const context = await browser.newContext();
+	const page = await context.newPage();
+	console.log(
+		"\nStarting to scrape https://www.takealot.com/all?sort=Relevance%EF%BB%BF%EF%BB%BF\n"
+	);
```
Comment (yanevdw) [inline src/index.ts:12]: Please remove.

File: src/index.ts [hunk 3]
```diff
@@ -1,2 +1,91 @@
...
+				productObject.rating = parseFloat(ratingAndReviews[0].slice(0, -1));
+				productObject.numberOfReviews = parseFloat(
+					ratingAndReviews[1].slice(0, -1)
+				);
+			}
+
+			scrapedProducts.push(productObject);
+		}
+		console.log("Products:", scrapedProducts);
+		console.log("Number of products scraped:", scrapedProducts.length);
```
Comment (yanevdw) [inline src/index.ts:84]: Please remove.

File: src/index.ts [hunk 4]
```diff
@@ -1,2 +1,91 @@
...
+	} catch (error) {
+		console.error("Error while scraping:", error);
+	} finally {
+		await browser.close();
+		console.log("\nWeb scraping ended X_X\n");
```
Comment (yanevdw) [inline src/index.ts:89]: Please remove.

**Review by yanevdw — APPROVED** [review]
Looks good, please just remove the console logs.

File: src/index.ts (hunk 4 above)
Comment (Meggie-H) [reply src/index.ts:89]: Thank you for all your comments. As far as I understand the "no console log" golden rule is does not necessarily apply to the backend. It's needed as we need to know what is happening in the backend and need to know if something went wrong. That said, it is much better practice to use a logger. I will ask Ayaaz to replace the console logs with log messages.

**Review by Julianna-Venter — APPROVED** [review]

**Pull request merged**